flink-ml-lib/src/main/java/org/apache/flink/ml/feature/kbinsdiscretizer/KBinsDiscretizerParams.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.feature.kbinsdiscretizer;
+
+import org.apache.flink.ml.param.IntParam;
+import org.apache.flink.ml.param.Param;
+import org.apache.flink.ml.param.ParamValidators;
+import org.apache.flink.ml.param.StringParam;
+
+/**
+ * Params for {@link KBinsDiscretizer}.
+ *
+ * @param <T> The class type of this instance.
+ */
+public interface KBinsDiscretizerParams<T> extends KBinsDiscretizerModelParams<T> {
+    String UNIFORM = "uniform";
+    String QUANTILE = "quantile";
+    String KMEANS = "kmeans";
+
+    /**
+     * Supported options to define the widths of the bins are listed as follows.
+     *
+     * <ul>
+     *   <li>uniform: all bins in each feature have identical widths.
+     *   <li>quantile: all bins in each feature have the same number of points.
+     *   <li>kmeans: values in each bin have the same nearest center of a 1D kmeans cluster.
+     * </ul>
+     */
+    Param<String> STRATEGY =
+            new StringParam(
+                    "strategy",
+                    "Strategy used to define the width of the bin.",
+                    QUANTILE,
+                    ParamValidators.inArray(UNIFORM, QUANTILE, KMEANS));
+
+    Param<Integer> NUM_BINS =
+            new IntParam("numBins", "Number of bins to produce.", 5, ParamValidators.gtEq(2));
+
+    Param<Integer> SUB_SAMPLES =
+            new IntParam(
+                    "subSamples",
+                    "Maximum number of samples used to fit the model.",
+                    200000,
+                    ParamValidators.gt(0));
```

Comment:
nit: Shall we add a check that `subSamples >= 2` or `subSamples >= numBins`?

Reply:
I agree that `subSamples < 2` is not a valid use case and I have udpated the code.

If `subSamples < numBins`, it should still work in `quantile` and `kmeans` case.